Core/src/ExponentialBackoff.php

```diff
@@ -112,9 +111,9 @@ public function execute(callable $function, array $arguments = [])
                 if ($this->retryListener) {
                     // Developer can modify the $arguments using the retryListener
                     // callback.
-                    $arguments = call_user_func_array(
+                    call_user_func_array(
                         $this->retryListener,
-                        [$exception, $retryAttempt, $arguments]
+                        [$exception, $retryAttempt, &$arguments]
```

Comment:
I modified this because a "listener" should not be expected to return a request and arguments. The ability to modify the request/arguments should be optional

Reply:
So, we did this change after the conversation about removing the call by ref arguments :)
Do you wanna go back the old route?

Reply:
Or do we change the name to something more appropriate thing like `modifyRequestFunction()`?

Reply:
Not really a conversation of this PR, but I'm not too convinced on the name `retryListener` anyway.
There is a `retryFunction` already present, and now we're adding a listener?
The difference isn't exactly clear.